code/modules/mob/living/simple_animal/hostile/megafauna/swarmer.dm

```diff
@@ -0,0 +1,205 @@
+#define MEDAL_PREFIX "Swarmer Beacon"
+#define TOO_MANY_SWARMERS_BEACON	10 //If there's this many swarmers, the beacon won't repopulate them
+#define TOO_MANY_SWARMERS 50 //Cap of AI swarmers, no reproduction past this number
+
+
+/*
+
+Swarmer Beacon
+
+A strange machine appears anywhere a normal lavaland mob can it produces a swarmer at a rate of
+1/15 seconds, until there are 10 swarmers, after this it is up to the swarmers themselves to
+increase their population (should they fall under 10, the machine will continue to repopulate back up to 10)
+
+tl;dr A million of the little hellraisers spawn (controlled by AI) and try to eat mining
+
+Loot: Not much, besides a shit load of artificial bluespace crystals, Oh and mining doesn't get eaten
+that's a plus I suppose.
+
+Difficulty: Special
+
+*/
+
+var/global/list/mob/living/simple_animal/hostile/swarmer/ai/AISwarmerMobs = list()
+
+/mob/living/simple_animal/hostile/megafauna/swarmer_swarm_beacon
+	name = "swarmer beacon"
+	desc = "That name is a bit of a mouthful, but stop paying attention to your mouth they're eating everything!"
+	icon = 'icons/mob/swarmer.dmi'
+	icon_state = "swarmer_console"
+	health = 2000
+	maxHealth = 2000 //low-ish HP because it's a passive boss, and the swarm itself is the real foe
+	medal_type = MEDAL_PREFIX
+	score_type = SWARMER_BEACON_SCORE
```

Comment:
No, if there's a medal or score it should be focused on the spawner, aye. Not sure if it needs a score at all though, but that's up to Kor.
Maybe it needs a sort of reactionary attack though. Like attacking it or getting near makes is spit out a bunch of extra hyper aggressive swarmers.